src/text.rs

```diff
@@ -179,6 +179,26 @@ impl<'a> Span<'a> {
             })
             .filter(|s| s.symbol != "\n")
     }
+
+    /// Applies a new style to an existing Span.
```

Comment:
It's probably worth noting that this is additive rather than replacing the style.

```suggestion
    /// Patch the style of the Span to be the combined style of the existing style plus the new style.
```